client/providers/google.go

```diff
@@ -120,12 +120,19 @@ func (g *GoogleOp) VerifyCICHash(ctx context.Context, idt []byte, expectedCICHas
 	return nil
 }
 
-func (g *GoogleOp) PublicKey(ctx context.Context, idt []byte) (crypto.PublicKey, error) {
-	j, err := jws.Parse(idt)
+func (g *GoogleOp) PublicKey(ctx context.Context, headersEnc []byte) (crypto.PublicKey, error) {
+	headersJSON, err := util.Base64DecodeForJWT(headersEnc)
 	if err != nil {
-		return nil, fmt.Errorf("failed to parse JWS: %w", err)
+		return nil, err
+	}
+
+	headers := make(map[string]any)
```

Comment:
Why would the headers not be `map[string]string`? Is something unmarshalling the strings into something else?